sr_port/op_unwind.c

```diff
@@ -102,11 +105,16 @@ void op_unwind(void)
 	assert(frame_pointer <= (stack_frame*)stackbase && frame_pointer > (stack_frame *)stacktop);
 	/* See if unwinding an indirect frame */
 	IF_INDR_FRAME_CLEANUP_CACHE_ENTRY(frame_pointer);
+	TREF(trig_forced_unwind) = trig_forced_unwind;	/* copy local back to global so "unw_mv_ent" can use that.
+							 * It is safe to do so as long as "unw_mv_ent" has no error codepath
+							 * which is true at this time.
+							 */
 	for (mvc = mv_chain; mvc < (mv_stent *)frame_pointer; )
 	{
 		unw_mv_ent(mvc);
 		mvc = (mv_stent *)(mvc->mv_st_next + (char *)mvc);
 	}
+	TREF(trig_forced_unwind) = FALSE;	/* reset this global right away in case we hit an error codepath below */
```

Comment:
The comment appears to be a cut/paste tag-along that's not appropriate here.

When does unw_mv_ent unwind the trigger mv_stent? Is it in this loop (when trig_forced_unwind is active)? If so, is this statement needed anyway since it is turned off in unw_mv_ent() at the end of the newly added block? Again if so, a comment that unw_mv_ent() will turn the flag off would probably be good though. Otherwise, nevermind.

Reply:
We do want to reset the global as soon as possible which is why it is done right after the unw_mv_ent() loop (it was set at the beginning of the unw_mv_ent loop). So the comment seems relevant to me. Can you clarify why it is not appropriate?

unw_mv_ent unwinds the trigger mv_sent in this loop. The FALSE reset is needed in op_unwind in case we do not find any MVST_TRGR mv_stent in the loop (in which case unw_mv_ent will not be doing the reset).

Reply:
There's an implicit understanding of what "right away" means at the top of the module. Down further, that meaning is lost (at least to me) - right away after what? I'd prefer a comment wording more like "Again, turn this global flag off while it is not needed in case of error below".